modules/primo/src/Primo/Ting/Object.php

```diff
@@ -84,37 +86,46 @@ public function getDescription() {
   /**
    * @inheritDoc
    */
-  public function getRelations() {
-    // TODO: Implement getRelations() method.
+  public function getAge() {
+    // TODO: Implement getAge() method.
   }
 
-
   /**
    * @inheritDoc
    */
-  public function getAge() {
-    // TODO: Implement getAge() method.
+  public function getAudience() {
+    // TODO: Implement getAudience() method.
   }
 
   /**
    * @inheritDoc
    */
-  public function getAudience() {
-    // TODO: Implement getAudience() method.
+  public function getContributors() {
+    return $this->document->getContributors();
   }
 
   /**
    * @inheritDoc
    */
-  public function getExtent() {
-    return $this->document->getDisplayFormat();
+  public function getCreators($format = self::NAME_FORMAT_DEFAULT) {
+    // Create a mapper for each name format. A mapper should take an array of
+    // elements for a name and combine them into a single string.
+    $defaultMapper = function(array $nameElements) {
+      return implode(' ', $nameElements);
+    };
+    $surnameFirstMapper = function(array $nameElements) {
+      return implode(', ', array_reverse($nameElements));
+    };
+    $mapper = ($format === self::NAME_FORMAT_SURNAME_FIRST) ? $surnameFirstMapper : $defaultMapper;
```

Comment:
Really like this pattern - takes an attempt or two to get used to, but from then on its great.